mod-source-record-manager-server/src/main/resources/templates/db_scripts/create_get_record_processing_log_function.sql

```diff
@@ -0,0 +1,99 @@
+-- Script to create function to get data import record processing log entries (recordProcessingLogDto).
+CREATE OR REPLACE FUNCTION get_record_processing_log(jobExecutionId uuid, recordId uuid)
+    RETURNS TABLE(job_execution_id uuid, source_id uuid, source_record_order integer, title text, source_record_action_status text, source_entity_id uuid, source_entity_hrid text, source_entity_error text, instance_action_status text, instance_entity_id uuid, instance_entity_hrid text, instance_entity_error text, holdings_action_status text, holdings_entity_id uuid, holdings_entity_hrid text, holdings_entity_error text, item_action_status text, item_entity_id uuid, item_entity_hrid text, item_entity_error text, order_action_status text, order_entity_id uuid, order_entity_hrid text, order_entity_error text, invoice_action_status text, invoice_entity_id uuid, invoice_entity_hrid text, invoice_entity_error text)
+AS $$
+BEGIN
+    RETURN QUERY EXECUTE format('
+        SELECT records_actions.job_execution_id,
+               records_actions.source_id,
+               records_actions.source_record_order,
+               rec_titles.title,
+               CASE WHEN marc_errors_number != 0 OR marc_actions[array_length(marc_actions, 1)] = ''NON_MATCH''
+                        THEN ''DISCARDED''
+                    WHEN marc_actions[array_length(marc_actions, 1)] = ''CREATE''
+                        THEN ''CREATED''
+                    WHEN marc_actions[array_length(marc_actions, 1)] IN (''UPDATE'', ''MODIFY'')
+                        THEN ''UPDATED''
+               END AS source_record_action_status,
+			   source_entity_id[1]::uuid,
+			   source_entity_hrid[1],
+			   source_entity_error[1],
+               get_entity_status(instance_actions, instance_errors_number) AS instance_action_status,
+			   instance_entity_id[1]::uuid,
+			   instance_entity_hrid[1],
+			   instance_entity_error[1],
+               get_entity_status(holdings_actions, holdings_errors_number) AS holdings_action_status,
+			   holdings_entity_id[1]::uuid,
+			   holdings_entity_hrid[1],
```

Comment:
let's return all array elements, it is assumed that multiple entities of the same type can be created/updated by the same source record